new na_ontap_motd module

dprts opened the pull request:
##### SUMMARY
This module allows setting/removing cluster and SVM level MOTD 

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
- na_ontap_motd
##### ANSIBLE VERSION
2.7

Spredzy committed:
scaleway: Create a base class for all the modules

Purpose of this commit is to create a base class for the upcoming
scaleway module to use.

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_security_group_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_volume_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_image_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_ip_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_server_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Spredzy committed:
scaleway: Introduce scaleway_snapshot_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

Commit:
na_ontap_motd module added

apollo13 committed:
Fixed hash_host option in known_hosts module. Fixes #44284

Commit:
Fix passwd function call

Spredzy committed:
scaleway: Introduce scaleway_organization_facts module

Co-authored-by: Rémy Leone <rleone@online.net>

remyleone committed:
scaleway: Refactor modules

KellerFuchs committed:
ldap_passwd.passwd_set: Use the synchronous API

Daniel-Sanchez-Fabregas committed:
In expect module documentation add `no_log` to password change example (#44436)

* Update expect.py

Recommend not showing passwords in the logs

+label: docsite_pr

* Fix unicode quotes

resmo committed:
doc: fix maintainer's name (#44452)

webknjaz committed:
Update readme info (#44394)

* Upgrade the number of contributors in README

* Upgrade the sponsor information in README

ganeshrn committed:
Update netconf_config module (#44379)

Fixes #40650
Fixes #40245
Fixes #41541

*  Refactor netconf_config module as per proposal #104
*  Update netconf_config module metadata to core network supported
*  Refactor local connection to use persistent connection framework
   for backward compatibility
*  Update netconf connection plugin configuration varaibles (Fixes #40245)
*  Add support for optional lock feature to Fixes #41541
*  Add integration test for netconf_config module
*  Documentation update
* Move deprecated options in netconf_config module

sivel committed:
Use newer is_sequence function instead of MutableSequence (#44331)

* Use newer is_sequence function instead of MutableSequence. Fixes #44327

* Add changelog for #44331

* Update changelog fragment to describe the fix in more detail

ruzickap committed:
Replacing old "with_items:" by "loop:" (#44303)

Replacing old "with_items:" by "loop:"

+label: docsite_pr

samccann committed:
Sumit's Alias add (#44467)

skylerbunny committed:
Return variable in get_url is 'status_code', not 'status' (#43366)

Subject is self explanatory.
Resolves #43365.

+label: docsite_pr

HuffAndPuff committed:
Documentation fix, decrypt is True by default (#44435)

andreaso committed:
Deal with case insensitivy in cloudflare_dns module (#44366)

Cloudflare's DNS API deals with the case insensitivy of DNS names and
IPv6 addresses by forcing them into lower case. To properly be able to
detect changes the Ansible module needs to behave accordingly.

To what extent the API expects sent DNS names to be lower case varies
between record types. Yet, since sending lower case always works, and
since we always get lower case back, it feels cleanest to use lower
case for all DNS names.

LindsayHill committed:
new nos_config module (#44140)

* new nos_config module

* Update cliconf/nos.py to add missing 'end'

LindsayHill committed:
nos_facts module and tests (#44094)

LindsayHill committed:
add missing slxos_config action plugin (#44103)

* add missing slxos_config action plugin

caphrim007 committed:
Adds the bigip_profile_http module (#44473)

This module can be used to manage HTTP profiles on a BIG-IP.

YunfanZhang42 committed:
Added Ansible Tower inventory plugin. (#41816)

Signed-off-by: Yunfan Zhang <yz322@duke.edu>

abadger committed:
Remove deprecated tags config option (#44479)

* Remove deprecated tags config option

* wordsmith porting guide entry

acozine via github

caphrim007 committed:
Removes dependencies and cleans up code (#44484)

Portions of the f5-sdk were removed as well as the netaddr library
and were replaced with libraries that are part of ansible. Additionally,
deprecated code has been removed.

caphrim007 committed:
Adds the bigip firewall DoS profile module (#44486)

This module can be used to manage AFM DoS profiles on a BIG-IP

caphrim007 committed:
Various bigiq fixes (#44487)

Fixes usage of the RestClient class. Documentation fixes. Removal
of dependency code.

jborean93 committed:
psrp: Added new Windows connection plugin (#41729)

* psrp: Added new Windows connection plugin

* Tweaks to connection options from review

bcoca committed:
fix tempating issues with no_log and loops (#44468)

* fix tempating issues with no_log and loops

 - task is no log if any item is
 - added test cases

fixes #43294

caphrim007 committed:
Correct unit tests and module fixes (#44488)

Cleanup of unit tests. Modules fixes in iapp_template nad profile_dns
to support unit tests.

mtregubov committed:
Update one_vm.py (#43349)

OpenNebula 5.5.8  expects int in DISK_SIZE field when you create a vm, not a float.
For example, "42 GB" should be sent to OpenNebula API as "43088", not as "43088.0". MEMORY you cast to int explicitly, but DISK_SIZE not.

Daniel-Sanchez-Fabregas committed:
Fix win_domain_computer (#44054)

Fixes #43653

- Change descrtiption parameter default to null.
  When reading an empty description from AD, powershell returns a null.
- Convert after and before diff keys to ordered hashes.
  Unordered state descriptions make --diff report equal values as
  changes.

pablorf-dev committed:
Fix handlers on strategy free (#33011)

Add _flushed_hosts dict to store when handlers are flushed and prevent them
to be executed too early using _filter_notified_hosts.

Add _wait_on_handler_results to wait only for handlers to be completed.

Remove only hosts that have been flushed from handler notified list.

Fix #31504, #23970

ganeshrn committed:
Fix network config line diff ignore (#43924)

* If config is added in form of lines add
  capability to ignore confgiure lines given as input
  in task

bcoca committed:
correct vault example (#44507)

* correct vault example

fixes #43897

* corrected text and code blocks

rambleraptor committed:
New GCP Module: gcp_compute_global_address_facts (#44029)

rambleraptor committed:
New GCP Module: gcp_compute_firewall_facts (#44027)

lingxiankong committed:
Add os_pool module for openstack load-balancing service (#43907)

amayer5125 committed:
Stat: doc: Change ignore MD5 example to generic checksum (#44480)

rambleraptor committed:
New GCP Module: gcp_compute_target_vpn_gateway (#42798)

sivel committed:
Raise a nicer error when we cannot execute the editor (#44423)

* Raise a nicer error when we cannot execute the editor. Fixes #44419

* Don't use to_bytes when constructing an exception

* Add changelog fragment

carchi8py committed:
Bug Fixes issue with netapp.py (#44283)

* Fix issue

gundalow reviewed (changes requested):
Needs review by someone in the NetApp org

carchi8py committed:
Bug Fixes for ontap_snapshot.py (#44216)

carchi8py committed:
Bug Fixes for ontap_net_vlan.py (#44209)

* Bug Fixes for ontap_net_vlan.py

* Make documentation changes

carchi8py committed:
Bug Fixes for ontap_net_routes.py (#44208)

carchi8py committed:
Bug Fixes for ontap_export_policy_rule.py (#44196)

* Bug Fixes for ontap_export_policy_rule.py

photoninger committed:
Fix for "cannot marshal None unless.." (#33660)

when using only an activation key without any channels.
As already suggested by mattclay in
https://github.com/ansible/ansible/pull/25079

and also patch unit test for rhn_register and
add test case for activationkey only

pilou- committed:
docker_container: add felixfontein as maintainer (#44426)

rambleraptor committed:
New GCP Module: gcp_compute_ssl_policy (#42797)

Christoph-D committed:
More consistent Docker unit parsing (#16748)

* Docker: Convert units for kernel_memory

Other memory arguments already use human_to_bytes to convert the
units.  This change makes the behavior more consistent.

* Update Docker unit parsing documentation

This explains the units more precisely and makes the documentation
more consistent.

This fixes #16526.

ymartin59 committed:
Update vault-keyring to ConfigManager internals (#41967)

Fix failure since ConfigManager changes

tyll committed:
service_facts: Use LC_ALL/LANG=C to run commands (#44474)

This allows to parse the output when the user's locale changes the
commands' output. For example chkconfig uses 'Ein' and 'Aus' instead of
'on' and 'off' when using LANG=de_DE.UTF-8 breaking the service
detection on RHEL 6.

bhuztez committed:
docker_container: fix working_dir (#42857)

WorkingDir should be read from Config instead of HostConfig, check
https://docs.docker.com/engine/api/v1.37/#operation/ContainerInspect

willthames committed:
Ensure that target group ports are cast to int on creation (#44442)

A previous fix fixed this for existing target groups but not
for new target groups

Fixes #37368

markgoddard committed:
Support project parameter for os_security_group module (#34472)

Many OpenStack modules provide the ability to register a resource in a
project other than the one being used to authenticate with, by adding a
project parameter to the module. Examples include os_network, os_subnet,
and os_router. This change adds a project parameter to the
os_security_group module.

Fixes: #34467
Fixes: #30292

stmarier committed:
CloudFormation: Allow failed changesets to gather stack_outputs (#43509)

benileo committed:
Fix spelling error of secondary_availability_zone in rds module (#43852)

pjrm committed:
Add 'instance_protocol' parameter when fetching ELB Listeners (#40785)

This PR changes the ec2_elb_facts module to output the value of ELB InstanceProtocol

wmosher committed:
fix ec2_vpc_peer example with incorrect parameter (#43905)

+label: docsite_pr

muxator committed:
doc: minor language corrections to loop docs (#42854)

"described more" -> "better described"
"more simple" -> "simpler"

+label: docsite_pr

carchi8py committed:
Adding ontap_vserver_peer.py Module (#44226)

* Adding ontap_vserver_peer.py Module

flawaetz committed:
Update playbooks_blocks.rst (#40951)

Blocks currently don't support loops such as with_items or sequence.  It would be helpful to make this clear in the docs otherwise it's a bit of a gotcha.

+label: docsite_pr

caphrim007 committed:
Removes netaddr dependency (#44519)

One more dep removed from bigip_selfip

pcahyna committed:
Describe subset and superset as "set theory tests" (#43926)

subset and superset do not belong to group theory but to set theory.

+label: docsite_pr

wookietreiber committed:
doc: fixes the the repetitions (#43937)

Wow, this does not seem to be an uncommon misspelling. Might be there
are some left that span over two lines. I noticed the one in the git
module and then used `grep -rw 'the the'` to find some more.

geekq committed:
Fixes #23078 - eliminate trailing whitespace in to_nice_json (#42633)

* Fix #23078 - eliminate trailing whitespace in to_nice_json

* Add changelog fragment for #42633

felixfontein committed:
ACME: improve tests (new Pebble version, strict mode) (#44518)

* Bump ACME test container version.

* Re-adding key rollover test.

pilou- committed:
Fix a typo in docs: s/patent/parent/ (#44529)

sivel committed:
Bump a few deprecations to correct versions (#44534)

felixfontein committed:
ACME: new helper module for ACME challenges which need TLS certs (#43756)

* Added helper module for generating ACME challenge certificates.

* Soft-fail on missing cryptography. Also check version.

* Adding integration test.

* Move acme_challenge_cert_helper from web_infrastructure to crypto/acme.

* Adjusting to draft-05.

* The cryptography branch has already been merged.

sdubrul committed:
added account_alias in the response of module aws_caller_facts (#42345)

* added account_alias in the response of module aws_caller_facts

* added comment to explain list_account_aliases

* renamed caller_identity to caller_facts as the content is extended

* created changelog

* security-policy needs the iam:ListAccountAliases for this module to work

* test now checks for the added field account_alias

* gracefully handle missing iam:ListAccountAliases permission

sivel committed:
Remove deprecated s3 module (#44537)

* Remove deprecated s3 module

* Add changelog

sivel committed:
Remove deprecated ec2_facts (#44536)

* Remove deprecated ec2_facts

* Add changelog

Commit:
adding postgresql server facts (#43328)

* adding postgresql server facts

* updating postgresql server facts, minor changes

* changed return dict to list

* fixed test

* fixed several issues

* several updates

* fixed naming

* list -> complex

* try again

* fixed mistake

* added user_visible_state to the result

* added fully_qualified_domain_name

* fixed second test

* updates

* fixed test

* updated samples in return value

* wait for postgresql server to be actually ready

* another approach for test

* second server was not created

* fixing pr comments

* fix sanity

* removed forgotten  ansible_facts

caphrim007 committed:
Fixes and features for bigip gtm monitor modules (#44540)

Merging in gtm monitor work from downstream repo. includes various
new fixes and features.

yuwzho committed:
Add route table and route module (#41175)

* add route module

* add test

* add table

* add  route table

* fix dict

* fix

* fix

* route  table accept no name

* add default

* fix

* fix

* fix

* fix pep

* support route table

* fix line ending

* fix pep

* fix

* fix

* set the default to 'None'

* make return value only id

* camel and snake

* set test alias

* change facts line ending

* change test

* fix

* add default

* fix

* fix line break

* remove unsafe args

* fix comment

* fix

caphrim007 committed:
Add time_wait_recyle param to bigip_profile_tcp (#44543)

Allows for configuring the time-wait-recycle setting on a tcp profile

abadger committed:
Fix new AnsiballZ code to always include module_utils/basic

The AnsiballZ optimization which only uses one pyton interpreter
currently monkeypatches the arguments into a global argument in module_utils
so we need to always include basic.py.  In the future we should fix this
so that it monkeypatches its own file.  That way we won't need to always
include basic.py

abadger committed:
Fixup removed and deprecated modules

* Removed modules no longer have documentation
  Decided this was causing people to think that modules were supported
  even after being removed.  This change is a new strategy to have the
  error message trying to use a removed module point people to the older
  documentation.

* Add stubs for modules removed in 2.7
  These are freshly removed so we want people who are still using them
  when they upgrade Ansible to have a hint as to where to find information
  on how to port.

* Finish properly undeprecating include
  include was undeprecated earlier but not all of the pieces that marked
  it as deprecated were reverted.  This change fixes the remaining
  pieces

caphrim007 committed:
Adds new params to bigip trunk (#44547)

* description and q-in-q

ryansb committed:
Extend `module_defaults` by adding default groups for cloud modules (#44127)

Extends `module_defaults` by adding a prefix to defaults `group/` which denotes a builtin list of modules. Initial groups are: `group/aws`, `group/azure`, and `group/gcp`

idahood committed:
Update shell.py (#44477)

* Update shell.py

<!--- Your description here -->
Add example for disabling warnings to shell module
+label: docsite_pr

* Fix PEP8 issues

guilhermesteinmuller committed:
Add missing changelog for PR #44418 (#44478)

The PR[1] was merged but we noticed that the
changelog file is missing.

[1] https://github.com/ansible/ansible/pull/44418

mdaniel committed:
Correct the default doc for attached in ec2_eni (#44523)

* Correct the default doc for attached in ec2_eni

Also corrected a typo in the summary

* Address ansible-test sanity error about E324

* Fix and remove the E325 suppression for ec2_eni

abadger committed:
Fix ansible-doc and docsite generation for removed modules

* Fix ansible-doc wrt removed modules
  * Fix listing of modules ia ansible-doc to not complain about removed modules
    Removed modules are marked as such in the metadata but nowhere else.
    Need to retrieve the metadata when a module doesn't have a doc so that
    we can tell if it falls under this case.
  * omit removed modules from json dump
  * Print an error that the module has been removed if attempting to run
    ansible-doc on that specific module

* Get plugin_formatter to stop outputting removed modules

justjais committed:
Includes support for Reverse-Mapping zone in nios_zone module (#44525)

* support reverse mapping

* support reverse mapping

* support reverse mapping

* fix shippable errors

* fix shippable errors

* fix shippable errors

justjais committed:
Added Support for excluding IPs in nios_next_ip and created new pluggin for nios_next_network (#44437)

* adding next network pluggin

* include exclude option for next_ip search

* changing example provider input

* adding new line at end

* version added info

* to fix shippable errors

* to fix shippable errors

* adding exclude option in doc section

* fix review comment

* fix review comment

felixfontein committed:
Clarify usage of env_file (#42211). (#44535)

jborean93 committed:
basic.py: catch ValueError when trying to import hash algorithms (#44551)

* basic.py: catch more Exceptions when trying to import md5 hash algorithms

andreaso committed:
Have nsupdate add expected quotation to TXT records (#43386)

This prevents the accidental creation of TXT records where every
single word gets split into its own string, such as TXT record values
in the format of `"foo" "bar" "baz"`. That being an implicit behavior
I have very hard to see anyone purposely relying on.

TXT record values can still explicitly be defined as one or more
strings, without any change in syntax.

Resolves #43380

guoqiao committed:
Fix mail module for python 3.7.0 (#44550) (#44552)

In python 3.7.0, changes in `ssl.py` breaks `smtplib.SMTP_SSL`, which
then breaks `mail` module in ansible.

Run this line in python shell:

    import smtplib;smtplib.SMTP_SSL().connect(host='smtp.gmail.com', port=465)

Before python 3.7.0, we will get:

    (220, b'smtp.gmail.com ESMTP j13-v6sm3086685pgq.56 - gsmtp')

In python 3.7.0, we get such error at `lib/python3.7/ssl.py` line 843, method `_create`:

    ValueError: server_hostname cannot be an empty string or start with a leading dot.

The ssl module is using host info on SMTP_SSL instance, which is not set.
The fix/workaround is simple, just pass host info to it:

    import smtplib;smtplib.SMTP_SSL(host='smtp.gmail.com').connect(host='smtp.gmail.com', port=465)

Fixes: #44550

Signed-off-by: Guo Qiao <guoqiao@gmail.com>

Commit:
Adding os_disk_size_gb parameter to Virtual Machine (#43316)

* added os disk size gb

* fixed mistake

* added os_disk_size_gb to test

* fixed test

* no default for disk size

ganeshrn committed:
Fix junos cli_command integration test (#44558)

* Issue rollback command before exit from config
  mode

vrischmann committed:
docker_container: allow arbitrary log_driver (#33579)

Docker supports logging plugin (see https://docs.docker.com/engine/admin/logging/plugins)
so it no longer makes sense to limit the log_driver option to a set of
fixed choices.

lingxiankong committed:
Add os_member module for openstack loadbalancing service (#43908)

dagwieers committed:
Add new contributor to team_ucs

trishnaguha committed:
mark cli_config as done on roadmap (#44567)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

machacekondra committed:
oVirt: Add depracation warning to all modules (#44440)

ganeshrn committed:
Add support for multiple prompt answers in network_cli (#44492)

* Currently network_cli support multiple prompts
  single answer as response. This PR adds support
  for multiple answers.
* In case of multiple prompts and mulitple answers the
  index of a particular prompt in the prompts list should
  match with the index in the answer list.

skylerbunny committed:
Update tags documentation to explain newer tags behavior (#44274)

LindsayHill committed:
Replace invalid_params with ValueError (#44545)

ganeshrn committed:
Add support for multiple answers in cli_command module (#44560)

* Add support for multiple answers in cli_command module

*  Add multiple answers support for multiple prompts

Depends on PR #44492

* Doc update

trishnaguha committed:
use retry_json nxos_banner (#44376)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

gdelpierre committed:
crypto: Add new module openssl_pkcs12 (#27320)

[module] Create new module to handle PKCS#12 file.

pilou- committed:
gather facts improvements (#44008)

* gather_subset is a list of strings

When gather_subset is an integer, a message pointing out the problem, current tb error 
is replaced by this one:

    ERROR! the field 'gather_subset' should be a list of (<class 'str'>,), but the item '42' is a <class 'int'>
...
* gathering_facts test: ensure smart gathering is on

jctanner committed:
New keyword: ignore_unreachable (#43857)

dagwieers committed:
docs: Use GitHub syntax highlighting styles (#44449)

This PR changes the CSS stylesheet so the documentation (and module)
examples look more like how they look on GitHub.

Lambdara committed:
Fix ignoring of 'Login Profile cannot be found' errors in iam (#43198)

ganeshrn committed:
Update 2.7 network roadmap done items (#44575)

samdoran committed:
Properly un expire account on creation (#44174)

When creating a new account, check to see if the expiration parameter is negative and pass in the appropriate parameter. Since the negative integer passed into expires is converted to time.struct_time which in turn gets converted to a formatted time string when passed to the underlying command, a -1 or large negative number would result in passing a date before 1970-01-01 to the underlying command.

This had the opposite effect of creating an account with no expiration account resulting in a newly created account that was already expired, or just throwing an error on certain systems.

sivel committed:
Remove bare_deprecated functionality (#44517)

* Remove bare_deprecated functionality

* Change tests due to bare_deprecated removal

samdoran committed:
Pass path to GalaxyRole object (#43051)

This will list all roles in all paths in roles_path rather than just the first path in roles_path.

rabru committed:
Update bigiq_application_https_offload.py (#44562)

<!--- Your description here -->

+label: docsite_pr

caphrim007 committed:
Minor fixes and enhancements to bigip modules (#44549)

Various cleanup and fixes in docs and iterations.

mattclay committed:
Use runtime instead of datetime for test output.

Commit:
adding app gateway module (#39940)

* adding app gateway module

* improved application gateway

* added register output

* subnet_output

* debugging

* dumping params

* fix dump

* fixed mapping

* fixed bug

* fixed param problem

* small fixes in code and added sample test

* fixed syntax

* fixing sanity stuff

* sanity fixes

* more sanity tests fixed

* added azure_tags

* tags

* pruning app gateway

* pruning app gateway

* string -> str

* removed type: str

* trying to add idempotency

* fixed syntax

* fix sanity

* fixed bugs

* fixed error

* disabled ssl policy check for  a while

* fixed dict comparison

* fixed syntax

* compare just http listeners & frontend ports

* fix syntax

* syntax

* syntax

* just compare sku

* fix syntax

* just http settings collections

* fixing appgw module

* make it green

* fix items issue

* compare all arrays

* fixed sanity

* adding snake_to_camel transformations

* fix

* another fix

* snake to camel

* additional snake to camel fixes

* fixed public ip address conversion

* version added is now 2.7

* changed alias

* dict_merge

* added ssl policy spec

* fixed new sanity issue

* renamed module

* fixed ssl policy translation

* added test for ssl & certificates

* updated doc

* fixed docs improved tests

* fix

* fixed test

amayer5125 committed:
Firewalld: Add bool Type To permanent In Docs (#44579)

sivel committed:
Bump sudo/su deprecation to 2.9 (#44581)

boriel committed:
Prevent utf-8 codec error in py3 (#44228)

Cannot upload templates in Python 3.x. Gives utf-8 codec error.
Fixes this by opening the template file in binary mode.

sivel committed:
Unit tests for ansible.playbook.become (#38541)

AlanCoding committed:
add missing entry to package_data (#44585)

caphrim007 committed:
Add parameter and remove netaddr (#44584)

This patch adds a description parameter to most fields. It also removes
netaddr from the modules.

sivel committed:
Deprecated version should be an explicit version number (#44593)

caphrim007 committed:
Misc fixes to bigip modules. (#44594)

Adding docs, formatting docs, adding validation checks and values

pilou- committed:
botmeta: scaleway isn't core (#44511)

GElkayam committed:
Fix assign_public_ip: no disregarded (#29691) (#43954)

in case assign_public_ip is False, it is not handled.

caphrim007 committed:
Removes f5-sdk code from bigip modules (#44595)

This patch is part of an ongoing effort to fully remove the f5sdk
from the f5 modules.

caphrim007 committed:
Adds features to device connectivity (#44598)

Adds new parameter and removes f5-sdk

Commit:
Fix docs errors (#33704)

Spredzy committed:
Vultr: Introduce the Vultr inventory plugin (#44400)

This commit introduce the Vultr inventory plugin.

felixfontein committed:
Adding module which allows to complete certificate chains (#44169)

* Adding certificate_complete_chain module.

* Avoid code smell errors.

* Removing input_chain_src.

* Make sure line ending is there.

sivel committed:
Pin deprecation of tags in params for include to deprecation of include (#44596)

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
I would make present the default

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
only FYI.
We found this not to be very useful, and we now prefer to assign the whole params dictionary to a single variable.

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
I don't know if the query element is really needed.  It seems superfluous as the API is already directed to this vserver.   (But we found other cases where the request seems over qualified).

Akasurde committed:
botmeta: IPA: Notify IPA team for module_utils changes (#44577)

Add team_ipa for changes in module_utils/ipa.py

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Ignore CentOS 6.10 for passwordstore test.

Spredzy committed:
BOTMETA: Create a $team_scaleway and add Spredzy in it (#44597)

dagwieers committed:
ACI: Add specific info about running locally (#43903)

Since the ACI modules (like most network-related modules) run on the
local controller, this PR adds the necessary details so users are aware
of this particular feature.

dagwieers committed:
Added an async 'started' test (like 'finished') (#43445)

s-hertel committed:
RDS inventory plugin (#41919)

Comments out uses of rds_instance in the integration tests and replace with AWS CLI until rds_instance has been merged

mattclay committed:
Fix PEP8 issue.

s-hertel committed:
Add diff mode for ec2_group (#44533)

* Add (preview) diff mode support ec2_group

* Add diff mode to some ec2_group integration tests

* Remove unnecessary arguments and add comment to the module notes

* Add changelog

bcoca committed:
Add new host_pinned strategy (#44586)

The 'free' strategy still attempts to do all hosts per task before going to the next, it just doesn't wait for slow hosts,
This strategy processes each host as fast as possible to the end of the play before trying to process another host in the pool.

brouberol committed:
Change DataDog to Datadog (#34273)

StephenSorriaux committed:
Add new parameters to manage mtime and atime for file module, fixes #30226 (#43230)

* Add new parameters related to mtime and atime for file module, fixes #30226

mattclay committed:
Split sanity tests into 3 groups.

willthames committed:
sns_topic boto3 port (#39292)

* Port sns_topic to boto3 and add tests

samdoran committed:
Add reboot action plugin (#35205)

* Update docs

* Add reboot action plugin

Refactor win_reboot so it is subclassed from reboot

* Use new connection methods

* Test fixes

* Use better uptime command for Linux

Use who -b to get the last time the system was booted rather than uptime, which changes every second.

* Use distribution specefic commands and flags

Query the managed node to determien its distribution, then set the appropriate command and flags.

* Tune debug messages a bit

* Update module docs with details about pre_reboot_delay

s docs

* Ensure that post_reboot_delay is a positive number

* Remove the stringification

* Add integration tests

* Make sure aliases are honored

* Handle systems that have an incorrect last boot time

SystemD and fakehw-clock do not properly set the
last boot time and instead always set it to epoch.
Use a different command if that is the case.

* Copyright and encoding fixes

* Minor fixes based on feedback

* Add exponential backoff to sucess check method

* Update integration test

Skip the integration test if it would try to reboot the control node. We need a new mechanism to account for this scenario in ansible-test, so tests must currently be run manually for this plugin.

* Update integration test

Skip the integration test if it would try to reboot the control node. We need a new mechanism to account for this scenario in ansible-test, so tests must currently be run manually for this plugin.

* Fail early with running with local connection

* Update docs based on feedback

* minor refactoring, state mgmt changes

abadger committed:
Fix another corner case of too many warnings for world readable current working directory

There should be no warning if there is no ansible.cfg file i nthe
current working directory.

abadger committed:
Remove from roadmap, thing that aren't going to make it for 2.7 (#44481)

* Remove from roadmap, thing that aren't going to make it for 2.7
* Added actual Windows roadmap deliverables

abadger committed:
Do not test for docs in removed modules

mattclay committed:
Move azure_rm_appgateway certs into files.

This avoids ~20 minutes of runtime on the no-tests-as-filters sanity test.

abadger committed:
Fix refs broken by removal of old module docs

abadger committed:
Fix unittests for always adding basic to the AnsiBallZ zips

Akasurde committed:
VMware: new module: vmware_local_role_facts (#44553)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

trishnaguha committed:
ignore file_pull (#44614)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

abadger committed:
Fix recursive_finder so it doesn't re-read files multiple times

This closes a corner case where recursive_finder could read a file
multiple times.

machacekondra committed:
ovirt_vm: Fix undefined variable (#44618)

Fix unassigned variable while suspending vm.

samerd committed:
issue:43021 add support for onyx version 3.6.6000 and above (#44527)

* issue:43021 add support for onyx version 3.6.6000

Signed-off-by: Samer Deeb <samerd@mellanox.com>

* issue:43021 add support for onyx version 3.6.6000

Signed-off-by: Samer Deeb <samerd@mellanox.com>

Akasurde committed:
Minor typo fix in package module documentation (#44617)

Added missing round bracket

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

abadger committed:
Change validate-modules for removed modules

Removed modules now don't have documentation.  Need to account for that
when checking them in validte-modules

amayer5125 committed:
Firewalld: Switch "true" To "yes" In Examples (#44588)

Matches documentation section values
Make module examples more consistent with core modules

Spredzy committed:
BOTMEA: Add  for lib/ansible/plugins/inventory/vultr.py (#44623)

ganeshrn committed:
Fix netconf_config source option issue (#44630)

ganeshrn committed:
Update netconf_config module doc and return diff (#44634)

* Update netconf_config module doc
* Return santizied before and after config
  in diff key

linnil1 committed:
[Junos] Add l3-interface option for vlan (#43863)

* [Junos] Add l3-interface option for vlan

* Add version_added for new option

YumaInaura committed:
Fix script module document (#42778)

script module seems does not create / remove files
it seems only checks file existence or not existence


+label: docsite_pr

Spredzy committed:
inventory/vultr: Few minor improvements (#44622)

idvoretskyi committed:
Fixed links to the CLI clients (#35792)

CLI clients link is incorrect, fixed.

ccoley committed:
Fix incorrect default value for shutdown_command (#44607)

dagwieers committed:
Various improvements to script module docs (#44637)

This PR includes:
- Use uppercase in descriptions
- Add trailing dot to descriptions
- Grammar/spelling fixes
- Adding names to examples as a best practice
- Simplify structure

victorock committed:
friendly message when requirement is not installed (#42321)

* friendly message when requirement is not installed

acdha committed:
ec2: add "IAM Role" to instance_profile_name

The AWS API and console docs are inconsistent about whether EC2 instances have IAM profiles or roles. Things which follow the API tend to use profile but the console uses “IAM Role”. This adds that term to the help text so it's searchable.

+label: docsite_pr

njferrant committed:
NetApp: return lun info for na_ontap_lun_map module (#42961)

* return lun details for na_ontap_lun_map

return lun details for na_ontap_lun_map.py and added return value documentation

pgaxatte committed:
Update os_server docs when removing an instance (#41963)

When removing an instance via its ID, it is not clearly explained that one should use the `name` parameter.

Also a simple example is provided.

+label: docsite_pr

lmprice committed:
Resolve issues in NetApp E-Series Host module (#39748)

* Resolve issues in NetApp E-Series Host module

The E-Series host module had some bugs relating to the update/creation
of host definitions when iSCSI initiators when included in the
configuration. This patch resolves this and other minor issues with
correctly detecting updates.

There were also several minor issues found that were causing issues with
truly idepotent updates/changes to the host definition.

This patch also provides some unit tests and integration tests to help
catch future issues in these areas.

fixes #28272

* Improve NetApp E-Series Host module testing

The NetApp E-Series Host module integration test lacked feature test
verification to verify the changes made to the storage array.

The NetApp E-Series rest api was used to verify host create, update, and
remove changes made to the NetApp E-Series storage arrays.

carchi8py committed:
Adding ontap_svm_options.py Module (#44220)

* Adding ontap_svn_options.py Module

carchi8py committed:
Adding ontap_software_update.py Module (#44218)

* Adding ontap_software_update.py Module

carchi8py committed:
Adding ontap_snapmirror.py Module (#44215)

* Adding ontap_snapmirror.py Module

carchi8py committed:
Adding ontap_fcp.py Module (#44197)

* Adding ontap_fcp.py Module

carchi8py committed:
Adding ontap_dns.py Module (#44193)

* Adding ontap_dns.py Module

* Change Short Description

carchi8py committed:
Adding ontap_cg_snapshot.py Module (#44181)

* Adding ontap_cg_snapshot.py Module

* Fix issues

carchi8py committed:
Adding ElementSW Volume Pair (#43984)

* Adding ElementSW Volume Pair

akira6592 committed:
fix variable names in vars file example (#44643)

carchi8py committed:
Adding ElementSW Volume Module (#43982)

* Adding ElementSW Volume Module

carchi8py committed:
Adding ElementSW Vlan Module (#43981)

* Adding ElementSW Vlan Module

carchi8py committed:
Adding ElementSW SnapShot Schedule Module (#43974)

* Adding ElementSW SnapShot Schedule Module

carchi8py committed:
Adding ElementSW Cluster Pair Module (#43967)

* Adding ElementSW Cluster Pair

carchi8py committed:
Adding ElementSW Backup Module (#43960)

* Adding ElementSW Backup Module

ptux committed:
Implement part of #27617 [expend checksum format to <algorithm>:(<checksum>|<url>)] (#43751)

* expend checksum format to <algorithm>:(<checksum>|<url>)

* continue to code at office

* ALPHA - expend checksum format to <algorithm>:(<checksum>|<url>)

* clean up tmpfile and comment

* try to add test code for 27617

* try to add test code for 27617

* try to add test code for 27617

* try to fix [Could not find or access 'testserver.py']

* fix test code [Could not find or access 'testserver.py']

* fix test code [add files dir]

* fix test code [files dir not exists]

* as [connection was closed before a valid response was received]

* [connection was closed before a valid response was received]

* [connection was closed before a valid response was received]

* add test item [sha1 and sha256]

* since [connection was closed before a valid response was received]

* fix [connection was closed before a valid response was received]

* fix test code typo

* add docs for #27617

* PR #43751 is minor change

* fix pep8 issue.

* fix test code style.

* fix unexpected quote

s-hertel committed:
Fix dangerous elb_application_lb state (#44646)

* elb_application_lb: fix dangerous default of deleting an ELB if state is omitted by changing state to default to present to be more like other AWS modules

arendej committed:
state: present added in docs (#37197)

* state: present added in docs

* atomic_container.py docs now match argspec, enforce test

lmprice committed:
Update the ignore and doc_fragments for E-Series

Make the generic updates to remove unneeded ignores and improve the
documentation for the NetApp E-Series modules.

caphrim007 committed:
Refactors device group (#44599)

Make module use patterns that other modules use

caphrim007 committed:
Misc fixes (#44649)

Correcting docs and adding a Parameters handler for description

caphrim007 committed:
Adds params and remove netaddr (#44648)

This patch adds new parameters to the vcmp guest module and removes
the netaddr python dependency

caphrim007 committed:
Adds new params to gtm server module (#44650)

Parameters to support iquery options were added.

caphrim007 committed:
Refactors bigip_iapp_service and adds params (#44653)

Module was refactored to look like other f5 modules. also added new
params.

* decription
* device_group
* metadata

caphrim007 committed:
Adds parameters and removes netaddr dependency (#44654)

several new parameters added to bigip_node and the netaddr dependency
has been removed.

caphrim007 committed:
Adds aliases parameter to gtm wide ip (#44652)

This parameter can assign aliases to a wide ip.

caphrim007 committed:
Removes more netaddr deps (#44657)

Removed dependency from bigip_snat_pool

caphrim007 committed:
Adds AS3 module (#44655)

This adds the as3 module for bigip to ansible.

dariko committed:
New module: docker_swarm_service (#19229)

ryansb committed:
Fix ec2_instance synopsis. Fixes #44641

caphrim007 committed:
Adds params to the profile-client-ssl module (#44656)

This patch adds new parameters to the client ssl module.

vista- committed:
Add example for configuration item deletion (#44390)

dagwieers committed:
Clean up BOTMETA.yml (#44574)

This PR includes:
- Removal of maintainers that are listed as author in the module
- Removal of entries that do not extend the original author list
- Move ignored-statement to namespace/directory (where useful)
- In some cases, fix the authors-list or add missing github id

We end up with a list of exceptions/additions and a large set of
namespace/directory maintainers or team of maintainers.

Some entries could be further improved by discussing with some
maintainers.

caphrim007 committed:
adds the bigip_cli_script module (#44674)

This module can be used to manage tmsh cli scripts on a BIG-IP

coreywan committed:
Added nios_txt_record module (#39264)

* Added nios_txt_record module

Whitespace cleanup and version fix

First stab at txt record integration test

Fix for CI version requirement

Added nios_txt_record module

Whitespace cleanup and version fix

First stab at txt record integration test

Fix for CI version requirement

force re-run

* added alias for cloud group1

Akasurde committed:
IPA: Add option to specify timeout (#44572)

This fix allows user to specify idle timeout for fetch_url used
internally in IPA connection and post_json call.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

augustkang committed:
doc: fixed typo (#44685)

felixfontein committed:
ACME: improve documentation (#44691)

* Override description for account_key_src and account_key_content to also mention private_key_*.

* Convert generic OpenSSL/cryptography remark from description to note.

This avoids the whole description list to be sorted alphabetically, which will be done by plugin_docs.py in case description is mentioned in both module fragment and module itself.

* Moving more notes to the notes: section.

* Uniformization of first paragraph. Mainly mention ACME supporting CAs, and only then mention Let's Encrypt as one of them.

* Adjusting to current drafts.

* Adjusting to updated drafts.

* Harmonizing short module descriptions.

* Referencing helper modules.

* Move general Let's Encrypt remark to doc fragment.

* Changing some Let's Encrypt references to more generic statements.

Daniel-Sanchez-Fabregas committed:
Module win_domain_computer fix delete computer with child (#44500)

* Module win_domain_computer fix delete computer with child

* add changelog fragment

crossan007 committed:
Switch to LiteralPath instead of Path. Closes #44508 (#44509)

* Switch to LiteralPath instead of Path. Closes #44508

* add changelog fragment

* fix line endings and remove final empty line

* Minor text changes in changelog

trishnaguha committed:
junos cli_config idempotence fix (#44706)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

lolcube committed:
fix typo (#44712)

denngie committed:
ios_vrf speed fix (#43765)

rvalle committed:
one_host environment variables, Fixes #44163 (#44568)

* including test case using environment variables as per issue #44163

* including missing environment variable in shared documentation fragement, related to issue #44163

* fixes parameters via environment variables, issue #44163

mfuchs committed:
Share the implementation of hashing for both vars_prompt and password_hash (#21215)

* Share the implementation of hashing for both vars_prompt and password_hash.
* vars_prompt with encrypt does not require passlib for the algorithms
  supported by crypt.
* Additional checks ensure that there is always a result.
  This works around issues in the crypt.crypt python function that returns
  None for algorithms it does not know.
  Some modules (like user module) interprets None as no password at all,
  which is misleading.
* The password_hash filter supports all parameters of passlib.
  This allows users to provide a rounds parameter, fixing #15326.
* password_hash is not restricted to the subset provided by crypt.crypt,
  fixing one half of #17266.
* Updated documentation fixes other half of #17266.
* password_hash does not hard-code the salt-length, which fixes bcrypt
  in connection with passlib.
  bcrypt requires a salt with length 22, which fixes #25347
* Salts are only generated by ansible when using crypt.crypt.
  Otherwise passlib generates them.
* Avoids deprecated functionality of passlib with newer library versions.
* When no rounds are specified for sha256/sha256_crypt and sha512/sha512_crypt
  always uses the default values used by crypt, i.e. 5000 rounds.
  Before when installed passlibs' defaults were used.
  passlib changes its defaults with newer library versions, leading to non
  idempotent behavior.

  NOTE: This will lead to the recalculation of existing hashes generated
        with passlib and without a rounds parameter.
        Yet henceforth the hashes will remain the same.
        No matter the installed passlib version.
        Making these hashes idempotent.

Fixes #15326
Fixes #17266
Fixes #25347 except bcrypt still uses 2a, instead of the suggested 2b.

* random_salt is solely handled by encrypt.py.
  There is no _random_salt function there anymore.
  Also the test moved to test_encrypt.py.
* Uses pytest.skip when passlib is not available, instead of a silent return.
* More checks are executed when passlib is not available.

* Moves tests that require passlib into their own test-function.

* Uses the six library to reraise the exception.

* Fixes integration test.

When no rounds are provided the defaults of crypt are used.
In that case the rounds are not part of the resulting MCF output.

pilou- committed:
BOTMETA.yml: remove superfluous labels (#44628)

Path components and filename without extension which are valid GitHub
labels are already default labels.

maxamillion committed:
YUM4/DNF compatibility via yum action plugin (#44322)

* YUM4/DNF compatibility via yum action plugin

DNF does not natively support allow_downgrade as an option, instead
that is always the default (not configurable by the administrator)
so it had to be implemented

 - Fixed group actions in check mode to report correct changed state
 - Better error handling for depsolve and transaction errors in DNF
 - Fixed group action idempotent transactions
 - Add use_backend to yum module/action plugin
 - Fix dnf handling of autoremove (didn't used to work nor had a
   default value specified, now does work and matches default
   behavior of yum)
 - Enable installroot tests for yum4(dnf) integration testing, dnf
   backend now supports that
 - Switch from zip to bc for certain package install/remove test
   cases in yum integration tests. The dnf depsolver downgrades
   python when you uninstall zip which alters the test environment
   and we have no control over that.
 - Add changelog fragment
 - Return a pkg_mgr fact if it was not previously set.

carchi8py commented:
@gundalow we are good with this one being merged in.

bcoca committed:
fix issue with no_log in py3

the error percolates differently from undefined vars in jinja2,
so fallback to use play_context as it is not involved in current templating

rrey committed:
modules/terraform: Quote the variable values in the command line (#43493)

* modules/terraform: Quote the variable values in the command line

Fixes: #43492

* fix six import

ckotte committed:
Fix systemd service is already masked issue (#44730)

* Check if service is already masked

Newer versions of Systemd now report a 'LoadError' when the unit file
is masked. This causes the play to fail with an error stating that the
service is already masked.

Now the systemd module checks if the service is masked and doesn't
fail if it's masked and LoadError is reported.

Fixes issue #42384.

* Remove useless parens

caphrim007 committed:
Adds bigip_software_install module (#44733)

This module can be used to install existing images on a bigip.

bcoca committed:
vars list (#43171)

* vars list

(cherry picked from commit 463b85cf0bcf7a557b857d1ebd3e8bf1b693fa4c)

* updated as per fb

maxamillion committed:
fix yum install with disablerepo (#44735)

Signed-off-by: Adam Miller <admiller@redhat.com>

caphrim007 committed:
Various fixes for f5 modules (#44734)

A number of bugfixes for the remaining 2.7 work on the F5 modules.

caphrim007 committed:
Adds metadata annotation that resource is managed by ansible (#44737)

Adds annotation so that other f5 utilities do not clobber config
by accident

YumaInaura committed:
Update playbooks_conditionals.rst (#43055)

+label: docsite_pr

annabunches committed:
Correct documentation for deployment_mode flag (#25978)

mattclay committed:
Download hello package from S3 for apt test.

mattclay committed:
Fix change detection on retried Shippable jobs.

willthames committed:
k8s_facts should not throw exceptions when not found (#44429)

Handle the case where a resource is not found by catching
the exception and returning an empty result set.

abadger committed:
Make links anonymous so that they don't trigger rstcheck

rstcheck flags duplicate link targets as errors

abadger committed:
Add changelog for 2.7

mattclay committed:
Get packages for dnf and yum tests from S3.

jainnikhil30 committed:
Add 'tower_settings' module for managing Ansible Tower Settings (#43933)

* add the tower_settings module

abadger committed:
Fix release to be correct

yungezz committed:
fix linux web app java linux_fx_version (#44715)

usaturn committed:
add_evaluation_delay of datadog monitor (#32784)

* add_evaluation_delay of datadog monitor

* fix version number

* fix version number

* update version num

abadger committed:
Add post modifier to version

Commit:
Revert "fix linux web app java linux_fx_version (#44715)" (#44750)

trishnaguha committed:
nxos_system idempotence fix (#44752)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

yungezz committed:
fix linux webapp java framework bug (#44753)

* fix linux web app java linux_fx_version

* fix lint

* fix test

* resolve comments

Commit:
Merge branch 'na_ontap_motd' of github.com:dprts/ansible into na_ontap_motd

Commit:
* state=present is now default

* assign whole params dictionary to a single variable

dprts commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
Agree, I am going to correct it.

dprts commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
Also I am going to apply this change

dprts commented on lib/ansible/modules/storage/netapp/na_ontap_motd.py:
It looks like here it is necessary but i will check it once more

dprts commented:
By mistake I pushed rebased branch. I am closing this and open new PR.
